plugin/src/main/java/org/opensearch/ml/action/models/UpdateModelTransportAction.java

```diff
@@ -269,10 +320,14 @@ private void updateModelWithRegisteringToAnotherModelGroup(
         String newModelGroupId,
         User user,
         MLUpdateModelInput updateModelInput,
-        ActionListener<UpdateResponse> actionListener,
-        ThreadContext.StoredContext context
+        ActionListener<UpdateResponse> wrappedListener,
+        boolean isModelDeployed
     ) {
         UpdateRequest updateRequest = new UpdateRequest(ML_MODEL_INDEX, modelId);
+        // This flag is used to decide if we need to re-deploy the predictor(model) when performing the in-place update
+        boolean isPredictorUpdate = (updateModelInput.getConnector() != null || updateModelInput.getConnectorId() != null);
+        // This flag is used to decide if we need to perform an in-place update
+        boolean isInPlaceUpdate = isModelDeployed && isPredictorUpdate;
```

Comment:
Are we performing in-place update for text embedding model too? Did we test for text embedding model too?